aiidalab_widgets_base/process.py

```diff
@@ -226,7 +369,9 @@ def __init__(self, process, **kwargs):
 
     def update(self):
         """Update the displayed output."""
-        calcs = get_running_calcs(self.main_process)
+        if self.process is None:
+            return
+        calcs = get_running_calcs(self.process)
```

Comment:
Since `get_running_calcs()` is now a generator function, you will need to refactor the code as described above. I would suggest the following pattern:
```python
calc = None
for calc in get_running_calcs(self.process):
    if calc.id == self.previous_calc_id:
        break
else:
    self.output = []
    # ...
if calc is not None and 'remote_folder' in calc.outputs:
    f_path = # ...
```
However, I'm actually a little bit confused by that. The `calc` chosen on line 238/383, is that just the (random) last calc produced by `get_running_calcs()`?

Reply:
I've no idea how you ended up in this outdated version of the file, but things have changed quite a bit since that version :).

Reply:
this is how it looks now:
https://github.com/aiidalab/aiidalab-widgets-base/blob/89234fb960e77f02f6b6efbc2dce9e7f4d039836/aiidalab_widgets_base/process.py#L437

and works pretty well I would say.